smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/httpResponse/HTTPResponseBindingOutputGenerator.kt

```diff
@@ -83,7 +83,14 @@ class HTTPResponseBindingOutputGenerator(
     private fun needsAReader(ctx: ProtocolGenerator.GenerationContext, responseBindings: List<HttpBindingDescriptor>): Boolean {
         return responseBindings
             .filter { !ctx.model.expectShape(it.member.target).hasTrait<StreamingTrait>() }
-            .any { it.location == HttpBinding.Location.PAYLOAD || it.location == HttpBinding.Location.DOCUMENT }
+            .any { hasPayloadThatNeedsReader(ctx, it) || it.location == HttpBinding.Location.DOCUMENT }
+    }
+
+    private fun hasPayloadThatNeedsReader(ctx: ProtocolGenerator.GenerationContext, binding: HttpBindingDescriptor): Boolean {
+        val targetShape = ctx.model.expectShape(binding.member.target)
+        return binding.location == HttpBinding.Location.PAYLOAD &&
+            (targetShape is StructureShape || targetShape is UnionShape) &&
+            !targetShape.members().isEmpty()
```

Comment:
This new logic in the `needsAReader()` function prevents creating unused readers (and the compiler warnings that come with them) by rendering reader creation only when a payload is a structure or union that has at least one member.

Other payloads (a string, an enum, a blob, smithy Document, etc.) do not require a reader.  Also a reader is not required for a structure or union that does not have at least one member.